site.js

```diff
@@ -8,21 +22,24 @@
  * @param {any} boxWidth
  * @param {any} boxHeight
  */
-function fitDimsToBounds(imgWidth, imgHeight, boxWidth, boxHeight) {
+function fitDimsToBounds(imgWidth, imgHeight, boxWidth, boxHeight, constrain) {
   var aspect = imgWidth / imgHeight;
+  var sizeMethod = constrain ? Math.min : Math.max;
   var dims = {
-    width: Math.min(boxWidth, imgWidth),
-    height: Math.min(boxHeight, imgHeight)
+    width: sizeMethod(boxWidth, imgWidth),
+    height: sizeMethod(boxHeight, imgHeight)
   };
 
   if (aspect > boxWidth / boxHeight) {
     // fit to width
-    dims.height = Math.round(dims.width / aspect);
+    dims.height = dims.width / aspect;
   } else {
     // fit to height
-    dims.width = Math.round(dims.height * aspect);
+    dims.width = dims.height * aspect;
   }
 
+  dims.width = Math.round(dims.width);
+  dims.height = Math.round(dims.height);
```

Comment:
Was there a reason you separated out these two steps (the operation and the math.round)? Seems like the previous code was a bit more compact.